### PR TITLE
NIAD-5: improve logging

### PR DESCRIPTION
**Pull request by bartek-sarul**

### CODE REVIEW

File: SCRWebService/main.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def main():
 if __name__ == "__main__":
     try:
         main()
-    except Exception as e:
-        logger.critical('001', 'Fatal exception in main application: {exception}', {'exception': e})
+    except:
+        logger.critical('Fatal exception in main application', exc_info=True)
```
Comment (grzegorz-robak) [inline SCRWebService/main.py:44]: Shouldn't `logger.exception()` be used here?

File: common/utilities/integration_adaptors_logger.py [hunk 2]
```diff
@@ -144,15 +67,43 @@ def _formatted_string(self, message: str, dict_values: dict) -> str:
...
-            (etype, value, trace) = exc_inf
-            if message:
-                return f'{message}, {traceback.format_exception(etype, value, trace)}'
-            else:
-                return f'{traceback.format_exception(etype, value, trace)}'
+
+class CustomFormatter(logging.Formatter):
+    def __init__(self):
+        super().__init__(
+            fmt='%(asctime)sZ | %(levelname)s | %(process)d | %(interaction_id)s | %(message_id)s | %(correlation_id)s | %(inbound_message_id)s | %(name)s | %(message)s',
```
Comment (grzegorz-robak) [inline common/utilities/integration_adaptors_logger.py]: Might be more readable to have that value as constant

File: common/utilities/integration_adaptors_logger.py [hunk 3]
```diff
@@ -144,15 +67,43 @@ def _formatted_string(self, message: str, dict_values: dict) -> str:
...
+            datefmt='%Y-%m-%dT%H:%M:%S.%f')
+
+    def format(self, record: LogRecord) -> str:
+        record.message_id = message_id.get() or ''
```
Comment (grzegorz-robak) [inline common/utilities/integration_adaptors_logger.py]: You should be able to use `record.message_id = message_id.get(default='')` or just specify default when each context variable is created (top of this file)  as doc on that method specifies:

> If there is no value for the variable in the current context, the method will:
         * return the value of the default argument of the method, if provided; or
         * return the default value for the context variable, if it was created
           with one; or
         * raise a LookupError.

File: common/utilities/integration_adaptors_logger.py [hunk 4]
```diff
@@ -13,121 +15,51 @@
...
-
-
-class IntegrationAdaptorsLogger(object):
+class IntegrationAdaptorsLogger(logging.LoggerAdapter):
     """
-    To log exceptions correctly, key them with 'exception'
+    Allows using dictonaries to format message
     """
+    def __init__(self, name: str):
+        super().__init__(logging.getLogger(name), None)
```
Comment (grzegorz-robak) [inline common/utilities/integration_adaptors_logger.py]: Might be more readable to name `None` parameter (`extra=None`)

File: mhs/common/mhs_common/workflow/synchronous.py [hunk 5]
```diff
@@ -69,42 +68,39 @@ def __init__(self,
                                                                                 interaction_details=interaction_details,
                                                                                 message=payload)
         except Exception:
-            logger.error('002', 'Failed to prepare outbound message')
+            logger.error('Failed to prepare outbound message')
```
Comment (grzegorz-robak) [inline mhs/common/mhs_common/workflow/synchronous.py:71]: `exception()` ?

File: mhs/common/mhs_common/workflow/synchronous.py [hunk 6]
```diff
@@ -69,42 +68,39 @@ def __init__(self,
...
             await wdo.set_outbound_status(wd.MessageStatus.OUTBOUND_MESSAGE_PREPARATION_FAILED)
             return 500, 'Failed message preparation', None
 
         if len(message) > self.max_request_size:
-            logger.error('004', 'Request to send to Spine is too large after serialisation. '
-                                '{RequestSize} {MaxRequestSize}',
-                         {'RequestSize': len(message), 'MaxRequestSize': self.max_request_size})
+            logger.error('Request to send to Spine is too large after serialisation. {RequestSize} {MaxRequestSize}',
```
Comment (grzegorz-robak) [inline mhs/common/mhs_common/workflow/synchronous.py:76]: `exception()` ?

File: mhs/common/mhs_common/routing/routing_reliability.py [hunk 7]
```diff
@@ -102,15 +102,19 @@ def __init__(self, spine_route_lookup_url: str, spine_org_code: str, client_cert
...
+                            "reliability_details": reliability_details
+                        })
             return reliability_details
-        except Exception as e:
-            logger.error("0006",
-                         "Couldn't obtain reliability details from Spine route lookup service for {org_code} & "
-                         "{service_id}. {exception}", {"org_code": org_code, "service_id": service_id, "exception": e})
-            raise e
+        except Exception:
+            logger.error("Couldn't obtain reliability details from Spine route lookup service for {org_code} & "
```
Comment (grzegorz-robak) [inline mhs/common/mhs_common/routing/routing_reliability.py]: `exception()` ?

**Review by grzegorz-robak — CHANGES_REQUESTED** [review]

File: mhs/common/mhs_common/workflow/synchronous.py (hunk 5 above)
Comment (bartek-sarul) [reply mhs/common/mhs_common/workflow/synchronous.py:71]: The original logging didn't log exception so I've left it the same

File: mhs/common/mhs_common/workflow/synchronous.py (hunk 6 above)
Comment (bartek-sarul) [reply mhs/common/mhs_common/workflow/synchronous.py:76]: The original logging didn't log exception so I've left it the same

File: SCRWebService/main.py (hunk 1 above)
Comment (bartek-sarul) [reply SCRWebService/main.py:44]: I didn't want to change logging level. Critical seems good here as the app crashes

**Review by jaroslaw-wojciechowski — DISMISSED** [review]

**Review by jaroslaw-wojciechowski — APPROVED** [review]

**Review by grzegorz-robak — APPROVED** [review]

**Pull request merged**